src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -509,77 +376,83 @@ public HttpResponseMessage Annotate(AnnotationPostedDataEntity annotateDocumentR
                 string documentType = annotateDocumentRequest.documentType;
+                string tempFilename = Path.GetFileNameWithoutExtension(documentGuid) + "_tmp";
+                string tempPath = Path.Combine(Path.GetDirectoryName(documentGuid), tempFilename + Path.GetExtension(documentGuid));
+
                 AnnotationDataEntity[] annotationsData = annotateDocumentRequest.annotationsData;
                 // initiate AnnotatedDocument object
                 // initiate list of annotations to add
-                List<AnnotationInfo> annotations = new List<AnnotationInfo>();
-              
-                DocumentInfoContainer documentInfo = AnnotationImageHandler.GetDocumentInfo(documentGuid, password);
-                // check if document type is image
-                if (SupportedImageFormats.Contains(Path.GetExtension(documentGuid)))
-                {
-                    documentType = "image";
-                }
-                // initiate annotator object  
-                string notSupportedMessage = "";
-                for (int i = 0; i < annotationsData.Length; i++)
+                List<AnnotationBase> annotations = new List<AnnotationBase>();
+
+                using (FileStream outputStream = File.OpenRead(documentGuid))
                 {
-                    // create annotator
-                    AnnotationDataEntity annotationData = annotationsData[i];
-                    PageData pageData = documentInfo.Pages[annotationData.pageNumber - 1];
-                    // add annotation, if current annotation type isn't supported by the current document type it will be ignored
-                    try
+                    using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(outputStream, GetLoadOptions(password)))
                     {
-                        BaseAnnotator annotator = AnnotatorFactory.createAnnotator(annotationData, pageData);
-                        if (annotator.IsSupported(documentType))
-                        {
-                            annotations.Add(annotator.GetAnnotationInfo(documentType));
-                        }
-                        else
+                        string notSupportedMessage = "";
+                        for (int i = 0; i < annotationsData.Length; i++)
                         {
-                            notSupportedMessage = annotator.Message;
+                            // create annotator
+                            AnnotationDataEntity annotationData = annotationsData[i];
+                            //PageData pageData = new PageData() { Height = 842, Width = 595 };
+                            IDocumentInfo info = annotator.Document.GetDocumentInfo();
+                            PageInfo pageInfo = info.PagesInfo[annotationsData[i].pageNumber];
+                            PageData pageData = new PageData() { Height = pageInfo.Height, Width = pageInfo.Width };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove these redundant parentheses.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372470)